lib/compat/wordpress-6.0/class-gutenberg-rest-block-patterns-controller.php

```diff
@@ -0,0 +1,154 @@
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base,
+			array(
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_items' ),
+					'permission_callback' => array( $this, 'get_items_permissions_check' ),
+				),
+				'schema'      => array( $this, 'get_public_item_schema' ),
+				'allow_batch' => array( 'v1' => true ),
+			)
+		);
+	}
+
+	/**
+	 * Checks whether a given request has permission to read navigation areas.
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 *
+	 * @return WP_Error|bool True if the request has read access, WP_Error object otherwise.
+	 */
+	public function get_items_permissions_check( $request ) { // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
+		if ( current_user_can( 'edit_posts' ) ) {
+			return true;
+		}
+
+		foreach ( get_post_types( array( 'show_in_rest' => true ), 'objects' ) as $post_type ) {
+			if ( current_user_can( $post_type->cap->edit_posts ) ) {
+				return true;
+			}
+		}
+
+		return new WP_Error(
+			'rest_cannot_view',
+			__( 'Sorry, you are not allowed to view the registered block patterns.', 'gutenberg' ),
+			array( 'status' => rest_authorization_required_code() )
+		);
+	}
+
+	/**
+	 * Retrieves all block patterns.
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 *
+	 * @return WP_Error|WP_REST_Response Response object on success, or WP_Error object on failure.
+	 */
+	public function get_items( $request ) { // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
+		// Load block patterns from w.org.
+		_load_remote_block_patterns(); // patterns with the `core` keyword
+		_load_remote_featured_patterns(); // patterns in the `featured` category
+		gutenberg_register_remote_theme_patterns(); // patterns requested by current theme
+
+		$response = array();
+		$patterns = WP_Block_Patterns_Registry::get_instance()->get_all_registered();
+		foreach ( $patterns as $pattern ) {
+			$prepared_pattern = $this->prepare_item_for_response( $pattern, $request );
+			$response[] = $this->prepare_response_for_collection( $prepared_pattern );
+		}
+		return rest_ensure_response( $response );
+	}
+
+	/**
+	 * Prepare a raw block pattern before it gets output in a REST API response.
+	 *
+	 * @param object          $item    Raw pattern as registered, before any changes.
+	 * @param WP_REST_Request $request Request object.
+	 * @return WP_REST_Response
+	 */
+	public function prepare_item_for_response( $item, $request ) {
+		$prepared_pattern = array(
+			'name'           => $item['name'],
+			'title'          => $item['title'],
+			'blockTypes'     => $item['blockTypes'],
+			'categories'     => $item['categories'],
+			'content'        => $item['content'],
+		);
+
+		$prepared_pattern = $this->add_additional_fields_to_object( $prepared_pattern, $request );
```

Comment:
```suggestion
                 $fields     = $this->get_fields_for_response( $request );
		$prepared_pattern = array();
		$keys = array( 'name', 'title', 'blockTypes', 'categories', 'content' );
		foreach( $keys as $key ) {
		    if ( rest_is_field_included( $key, $fields ) ) {
			$prepared_pattern[$key] = $item[$key];
		     }
		}

		$prepared_pattern = $this->add_additional_fields_to_object( $prepared_pattern, $request );
```

Let's try this.

Reply:
@jsnajdr This still needs eyes on.

Reply:
The `_fields` list is already being honored, just in a different way, I described it here: https://github.com/WordPress/gutenberg/pull/39185#issuecomment-1070924513

Or is it preferred that `rest_is_field_included` is used explicitly by all new endpoints?